R/plot_cluster_sina.R

```diff
@@ -0,0 +1,133 @@
+#' Create a sina plot against lineage
+#'
+#' Side-effect: this function creates a plot and stores it to a file.
+#'
+#' @param pldf  Data-frame. The data element of a tree plot.
+#' @param output_dir  String. The directory within which the plot should be stored.
+#' @param varx  String. Which variable in \code{pldf} should be plotted on the y-axis.
+#' @param mut_regexp  String. Regular-expression(s) for restricting the plot to a subset of the rows
+#'   in \code{pldf}. Only rows with an "allmuts" entry that matches one of these regular expressions
+#'   will be presented. If NULL, all rows are presented.
+#' @param lineage_regexp  String. Regular-expression(s) for restricting the lineages that are
+#'   presented in the plot. Only those rows of \code{pldf} with a "lineage" entry that matches one
+#'   of these regular expressions will be presented. If NULL, all rows are presented.
+
+plot_cluster_sina <- function(pldf,
+                              output_dir,
+                              varx = "logistic_growth_rate",
+                              mut_regexp = "S:A222V",
+                              lineage_regexp = NULL) {
+  sc1 <- format_cluster_sina_data(
+    pldf,
+    varx = varx,
+    mut_regexp = mut_regexp,
+    lineage_regexp = lineage_regexp
+  )
+
+  p1 <- create_cluster_sina_ggplot(sc1, y_lab = varx)
+
+  g1 <- create_widget(
+    ggobj = p1,
+    width_svg = 8,
+    height_svg = 8
+  )
+
+  htmlwidgets::saveWidget(g1,
```

Comment:
Happy to leave this in for now as well to make changes step-by-step - though this line will be unnecessary when we save the ggiraph objects as .rds files.